php/class-wp-app-manifest.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * WP_APP_Manifest class.
+ *
+ * @package PWA
+ */
+
+/**
+ * WP_APP_Manifest class.
+ *
+ * Mainly copied from Jetpack_PWA_Manifest and Jetpack_PWA_Helpers.
+ */
+class WP_APP_Manifest {
+
+	/**
+	 * The theme color to use if no dynamic value is present.
+	 *
+	 * @var string
+	 */
+	const FALLBACK_THEME_COLOR = '#fff';
+
+	/**
+	 * The REST API namespace for the manifest request.
+	 *
+	 * @var string
+	 */
+	const REST_NAMESPACE = 'wp/v2';
+
+	/**
+	 * The REST API route for the manifest request.
+	 *
+	 * @var string
+	 */
+	const REST_ROUTE = '/pwa-manifest';
+
+	/**
+	 * The default manifest icon sizes.
+	 *
+	 * Copied from Jetpack_PWA_Helpers::get_default_manifest_icon_sizes().
+	 * Based on a conversation in https://github.com/GoogleChrome/lighthouse/issues/291
+	 *
+	 * @var int[]
+	 */
+	public $default_manifest_icon_sizes = array( 192, 512 );
+
+	/**
+	 * Inits the class.
+	 *
+	 * Mainly copied from Jetpack_PWA_Manifest::__construct().
+	 */
+	public function init() {
+		add_action( 'wp_head', array( $this, 'manifest_link_and_meta' ) );
+		add_action( 'rest_api_init', array( $this, 'register_manifest_rest_route' ) );
+	}
+
+	/**
+	 * Outputs the <link> and <meta> tags for the app manifest.
+	 *
+	 * Mainly copied from Jetpack_PWA_Manifest::render_manifest_link().
+	 */
+	public function manifest_link_and_meta() {
+		?>
+		<link rel="manifest" href="<?php echo esc_url( rest_url( self::REST_NAMESPACE . self::REST_ROUTE ) ); ?>">
+		<meta name="theme-color" content="<?php echo esc_attr( $this->get_theme_color() ); ?>">
+		<?php
+	}
+
+	/**
+	 * Gets the theme color for the manifest.
+	 *
+	 * Mainly copied from Jetpack_PWA_Helpers::get_theme_color().
+	 * First looks for the header background color in the AMP for WordPress plugin, if it's active.
+	 * This color displays on loading the app.
+	 *
+	 * @return string $theme_color The theme color for the manifest.json file, as a hex value.
+	 */
+	public function get_theme_color() {
+		if ( current_theme_supports( 'custom-background' ) ) {
+			$background_color = get_background_color(); // This returns a hex value without the leading #, or an empty string.
+			if ( $background_color ) {
+				$theme_color = "#{$background_color}";
+			}
+		}
+
+		if ( ! isset( $theme_color ) ) {
+			$theme_color = self::FALLBACK_THEME_COLOR;
+		}
+
+		/**
+		 * Enables overriding the PWA theme color.
+		 *
+		 * @param string $theme_color Hex color value.
+		 */
+		return apply_filters( 'pwa_background_color', $theme_color );
```

Comment:
Instead of adding a filter for this, we can just let the color be filtered via the `pwa_manifest_json` filter.